contentcuration/contentcuration/viewsets/contentnode.py

```diff
@@ -568,22 +570,30 @@ def requisites(self, request, pk=None):
             ),
         )
 
-    @detail_route(methods=["get"])
+    @action(detail=True, methods=["get"])
     def size(self, request, pk=None):
         if not pk:
             raise Http404
+
+        task_info = None
         node = self.get_object()
-        files = (
-            File.objects.filter(
-                contentnode__in=node.get_descendants(include_self=True),
-                contentnode__complete=True,
+
+        size, stale = calculate_resource_size(node=node, force=False)
+        if stale:
+            task_args = dict(node_id=node.pk, channel_id=node.channel_id)
+            task_info = get_or_create_async_task(
```

Comment:
Because it took me all afternoon to realize it - might be good to add a comment here that `get_or_create_async_task` is being used here to prevent a cache stampede.